addons/mail/models/mail_channel_partner.py

```diff
@@ -135,6 +135,22 @@ def _rtc_leave_call(self):
         else:
             return self.channel_id._rtc_cancel_invitations(partner_ids=self.partner_id.ids, guest_ids=self.guest_id.ids)
 
+    def _rtc_sync_sessions(self, check_rtc_session_ids=None):
+        """Synchronize the RTC sessions for self channel partner.
+            - Inactive sessions of the channel are deleted.
+            - Current sessions are returned.
+            - Sessions given in check_rtc_session_ids that no longer exists
+              are returned as non-existing.
+        Returns a list of JS command for updating rtcSessions field of channel.
+        """
+        self.ensure_one()
+        self.channel_id.rtc_session_ids._delete_inactive_rtc_sessions()
+        check_rtc_sessions = self.env['mail.channel.rtc.session'].browse([int(check_rtc_session_id) for check_rtc_session_id in (check_rtc_session_ids or [])])
+        return [
+            ('insert', [rtc_session_sudo._mail_rtc_session_format() for rtc_session_sudo in self.channel_id.rtc_session_ids]),
+            ('insert-and-unlink', [{'id': missing_rtc_session_sudo.id} for missing_rtc_session_sudo in check_rtc_sessions - self.channel_id.rtc_session_ids]),
+        ]
```

Comment:
IMO not super fond of this method that both acts as getter for RTC sessions and side-effect cleaning inactive sessions. The naming of method feels off because it doesn't look the return value is meaningful when it's actually the most important aspect. And it's not good for a get method to have side-effects.